py/client/pydeephaven/session.py

```diff
@@ -206,12 +287,23 @@ def run_script(self, script: str) -> None:
         Raises:
             DHError
         """
+
         with self._r_lock:
+            if self._sync_fields == SYNC_REPEATED:
+                self._cancel_update_fields()
+            
             response = self.console_service.run_script(script)
-            self._parse_script_response(response)
+
+            if self._sync_fields == SYNC_REPEATED:
+                # We can ignore the script response because
+                # all the new tables are added by this call anyways
+                self._fields = {}
+                self.sync_fields(repeating=True)
```

Comment:
So this isn't actually true if there are other list field subscribers at the time run_script is called. If so, run_script changes get batched with a certain frequency (250ms by default), so they aren't necessarily visible on the first return from sync_fields. I think just applying both will "work".